R/schematic_rest_api.R

```diff
@@ -0,0 +1,160 @@
+  
+  manifest_url <- httr::content(req)
+  manifest_url
+}
+
+
+#' schematic rest api to validate metadata
+#' 
+#' @param url URL to schematic API endpoint
+#' @param schema_url URL to a schema jsonld 
+#' @param data_type Type of dataset
+#' @param csv_file Filepath of csv to validate
+#' 
+#' @returns An empty list() if sucessfully validated. Or a list of errors.
+#' @export
+manifest_validate <- function(url="http://localhost:3001/v1/model/validate",
+                              schema_url="https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld", #nolint
+                              data_type, csv_file) {
+  req <- httr::POST(url,
+                     query=list(
+                       schema_url=schema_url,
+                       data_type=data_type),
+                     body=list(csv_file=httr::upload_file(csv_file))
+  )
+  
+  annotation_status <- httr::content(req)
+  annotation_status
+}
+
+
+#' schematic rest api to submit metadata
+#' 
+#' @param url URL to schematic API endpoint
+#' @param schema_url URL to a schema jsonld 
+#' @param data_type Type of dataset
+#' @param dataset_id Synapse ID of existing manifest
+#' @param input_token Synapse login cookie, PAT, or API key.
+#' @param csv_file Filepath of csv to validate
+#' 
+#' @returns TRUE if successful upload or validate errors if not.
+#' @export
+model_submit <- function(url="http://localhost:3001/v1/model/submit",
+                         schema_url="https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld", #notlint
+                         data_type, dataset_id, input_token, csv_file) {
+  req <- httr::POST(url,
+                    #add_headers(Authorization=paste0("Bearer ", pat)),
+                    query=list(
+                      schema_url=schema_url,
+                      data_type=data_type,
+                      dataset_id=dataset_id,
+                      input_token=input_token),
+                    body=list(csv_file=httr::upload_file(csv_file))
+  )
+  
+  manifest_id <- httr::content(req)
+  manifest_id
+}
+
+#' Gets all datasets in folder under a given storage project that the current user has access to.
+#' 
+#' @param url URL to schematic API endpoint
+#' @param syn_master_file_view synapse ID of master file view.
+#' @param syn_master_file_name Synapse storage manifest file name.
+#' @param project_id synapse ID of a storage project.
+#' @param input_token synapse PAT
+#'
+#'@export
+storage_project_datasets <- function(url="http://localhost:3001/v1/storage/project/datasets",
+                                     syn_master_file_view,
+                                     syn_master_file_name,
+                                     project_id,
+                                     input_token) {
+  
+  req <- httr::GET(url,
+                    #add_headers(Authorization=paste0("Bearer ", pat)),
+                    query=list(
+                      syn_master_file_view=syn_master_file_view,
```

Comment:
These parameters will change. We'll sync up when the REST API calls are finalized.